src/mondo_robot_template.py

```diff
@@ -21,31 +21,98 @@
-    df_medgen_mondo = df[df['source'] == 'MONDO'][['source_id', 'xref_id']].rename(columns={'source_id': 'mondo_id'})
-    out_df_cui_cn = df_medgen_mondo.copy()
-    out_df_cui_cn['xref_id'] = out_df_cui_cn['xref_id'].apply(add_prefixes_to_plain_id)
+    df = get_mapping_set(input_file, add_prefixes=True, sssomify=False, filter_out_medgencui=filter_out_medgencui)
+
+    # Mondo->MEDGEN & Mondo->UMLS
+    # 1. Get explicit Medgen (CUI, CN) -> Mondo mappings
+    df_umls_mondo = df[df['source'] == 'MONDO'][['source_id', 'xref_id']].rename(
+        columns={'source_id': 'mondo_id', 'xref_id': 'umls_cui'})
 
-    # Get Medgen (UID) -> Mondo mappings
+    # 2. Get Medgen (UID) -> Mondo mappings
     # - Done by proxy: UID <-> CUI <-> MONDO
-    df_medgen_medgenuid = df[df['source'] == 'MedGen'][['source_id', 'xref_id']].rename(
-        columns={'source_id': 'medgen_uid'})
-    # todo: should some of these steps be in _reformat_mapping_set()? to be utilized by SSSOM files?
-    out_df_uid = pd.merge(df_medgen_mondo, df_medgen_medgenuid, on='xref_id').rename(
-        columns={'xref_id': 'source_id', 'medgen_uid': 'xref_id'})[['mondo_id', 'xref_id', 'source_id']]
-    out_df_uid['xref_id'] = out_df_uid['xref_id'].apply(lambda x: f'MEDGEN:{x}')
-    out_df_uid['source_id'] = out_df_uid['source_id'].apply(lambda x: f'UMLS:{x}')
+    df_umls_medgenuid = df[df['source'] == 'MedGen'][['source_id', 'xref_id']].rename(
+        columns={'source_id': 'medgen_uid', 'xref_id': 'umls_cui'})
+    df_umls_medgenuid['medgen_uid'] = (
+        df_umls_medgenuid['medgen_uid'].apply(add_prefixes_to_plain_id))  # should/will all be MEDGEN
+    df_merged = pd.merge(df_umls_mondo, df_umls_medgenuid, on='umls_cui')
+    # - Split into (Mondo <-> Medgen UID) & (Mondo <-> UMLS CUI)
+    out_df_medgenuid = df_merged.rename(columns={'medgen_uid': 'xref_id', 'umls_cui': 'source_id'})[[
+        'mondo_id', 'xref_id', 'source_id']]
+    out_df_medgenuid['source_id'] = ''
+    out_df_umlscui = df_merged.rename(columns={'umls_cui': 'xref_id', 'medgen_uid': 'source_id'})
+
+    # Mondo->MESH
+    df_umls_mesh = get_mapping_set(input_file, filter_sources=['MeSH'], add_prefixes=True, sssomify=False)
+    df_umls_mesh['source_id'] = df_umls_mesh['source_id'].apply(lambda x: 'mesh:' + x)
+    out_df_mesh = pd.merge(df_umls_mesh, df_umls_mondo, left_on='xref_id', right_on='umls_cui').rename(
+        columns={'source_id': 'xref_id', 'xref_id': 'source_id'})[['mondo_id', 'xref_id', 'source_id']]
+
+    # Combine mappings
+    out_df = pd.concat([out_df_medgenuid, out_df_umlscui, out_df_mesh]).sort_values(['xref_id', 'mondo_id'])\
+        .drop_duplicates().fillna('')
+
+    # Add additional cols
+    out_df['source_medgen_id'] = 'MONDO:MEDGEN'
+    # todo: could optimize by doing apply just on the xref_id col
+    def set_mapping_pred(row):
+        """Set mapping predicate"""
+        # M = Concept UI; D = Descriptor UI; C = SupplementalRecordUI; Q = Qualifier UI
```

Comment:
## No illegal proxy merges present
After handling determining that there are [no singular MeSH concepts in the MedGen matchings](https://github.com/monarch-initiative/medgen/pull/19/files#r1565190951), it follows that there are no fortunately no illegal proxy merges.

Reply:
yes, i believe a proxy merge (or the qc for it) is based on the xref axiom `Mondo:equivalentTo` and since all of these are being added with `Mondo:relatedTo` there is not longer a possibility for these additions to cause a proxy merge.